integration-tests/tests/src/tests/enums.ts

```diff
@@ -0,0 +1,92 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { expect } from "chai";
+import {
+  ConnectionState,
+  NumericLogLevel,
+  OpenRealmBehaviorType,
+  OpenRealmTimeOutBehavior,
+  SessionStopPolicy,
+} from "realm";
+
+describe("Enums", function () {
+  describe("ConnectionState", function () {
+    it("is accessible", function () {
+      expect(ConnectionState).to.be.an("object");
+      expect(ConnectionState).to.deep.equals({
+        Disconnected: "disconnected",
+        Connecting: "connecting",
+        Connected: "connected",
+      });
+    });
+  });
+  describe("SessionStopPolicy", function () {
+    it("is accessible", function () {
+      expect(SessionStopPolicy).to.be.an("object");
+      expect(SessionStopPolicy).to.deep.equal({
+        AfterUpload: "after-upload",
+        Immediately: "immediately",
+        Never: "never",
+      });
+    });
+  });
+  describe("OpenRealmBehaviorType", function () {
+    it("is accessible", function () {
+      expect(OpenRealmBehaviorType).to.be.an("object");
+      expect(OpenRealmBehaviorType).to.deep.equal({
+        DownloadBeforeOpen: "downloadBeforeOpen",
+        OpenImmediately: "openImmediately",
+      });
+    });
+  });
+  describe("OpenRealmTimeOutBehavior", function () {
+    it("is accessible", function () {
+      expect(OpenRealmTimeOutBehavior).to.be.an("object");
+      expect(OpenRealmTimeOutBehavior).to.deep.equal({
+        OpenLocalRealm: "openLocalRealm",
+        ThrowException: "throwException",
+      });
+    });
+  });
+  describe("NumericLogLevel", function () {
+    it("is accessible", function () {
+      expect(NumericLogLevel).to.be.an("object");
+      expect(NumericLogLevel).to.deep.equals({
```

Comment:
This first line shouldn't be needed when you're doing deep equals to an object. Btw, does it still work with having the `s`at the end of `equal`? (I'd vote for where you're using `to.deep.equal()`) :)

Reply:
Yeah, most of these "to" "be" "equal/equals" are just meaningless syntactic sugar that does not change its behavior. So it could be `expect(NumericLogLevel).deep.equals` or `expect(NumericLogLevel).to.deep.equal`. I like both, I have generally been using the former for less use of this chaining but the latter does read nicer.